ericblade/quagga2
# WIP: begin significant updates to test suite structure (fix #75)

- add ts-mocha
- add several new npm test scripts:
  test:browser-specific, test:browser-universal, test:node-specific,
  test:node-universal, test:browser-all, test:node-all, test:module
- add an initial implementation for the node tests
- npm test script runs module tests (module import and require), all node
  tests, and the original PhantomJS test suite
- new test scripts will be located in a 'test' directory off of the
  directory that contains the module being tested. tests that should
  perform equally in browser and node will be in the main 'test'
  directory, while browser and node specific tests will have their own
  directories.
  Ergo, the structure should eventually look something like:
  - src/common/test - universal tests for modules in src/common
  - src/common/test/node - node specific tests for modules in src/common
  - src/common/test/browser - browser specific tests for modules in
    src/common

- updates to browser based testing will be in a future changeset
- Events tests (test/events.spec.ts) moved to
  src/common/test/events.spec.ts
- Added unit test for src/quagga/transform as
  src/quagga/test/transform.spec.ts
- test/tsconfig.json is still used as the base tsconfig for running tests.
  It is now running with module type 'commonjs' to avoid odd
  "import can only be used in a module" types of errors.
- add mocha to types in main tsconfig, to prevent tests from appearing as
  files full of errors in editor
